examples/step-71/step-71.cc

```diff
@@ -0,0 +1,937 @@
+                       * jump_grad_j_dot_n      //   [grad u n]
+                                                //
+                     - av_hessian_j_dot_n_dot_n // - {grad^2 u n n}
+                         * jump_grad_i_dot_n    //   [grad v n]
+                                                //
+                     + gamma                    //  gamma
+                         * jump_grad_i_dot_n    // [grad v n]
+                         * jump_grad_j_dot_n    // [grad u n]
+                     ) *
+                    JxW[qpoint]; // dx
+                }
+
+              copy_data.cell_rhs(i) +=
+                (-av_hessian_i_dot_n_dot_n *       // - {grad^2 v n n }
+                   (exact_gradients[qpoint] * n)   //   (grad u_exact . n)
+                 +                                 // +
+                 gamma                             //  gamma
+                   * jump_grad_i_dot_n             // [grad v n]
+                   * (exact_gradients[qpoint] * n) // (grad u_exact . n)
+                 ) *
+                JxW[qpoint]; // dx
+            }
+        }
+    };
+
+    // Part 4 is a small function that copies the data produced by the
+    // cell, interior, and boundary face assemblers above into the
+    // global matrix and right hand side vector. There really is not
+    // very much to do here: We distribute the cell matrix and right
+    // hand side contributions as we have done in almost all of the
+    // other tutorial programs using the constraints objects. We then
+    // also have to do the same for the face matrix contributions
+    // that have gained content for the faces (interior and boundary)
+    // and that the `face_worker` and `boundary_worker` have added
+    // to the `copy_data.face_data` array.
+    auto copier = [&](const CopyData &copy_data) {
+      constraints.distribute_local_to_global(copy_data.cell_matrix,
+                                             copy_data.cell_rhs,
+                                             copy_data.local_dof_indices,
+                                             system_matrix,
+                                             system_rhs);
+
+      for (auto &cdf : copy_data.face_data)
+        {
+          constraints.distribute_local_to_global(cdf.cell_matrix,
+                                                 cdf.joint_dof_indices,
+                                                 system_matrix);
+        }
+    };
+
+
+    // Having set all of this up, what remains is to just create a scratch
+    // and copy data object and call the MeshWorker::mesh_loop() function
+    // that then goes over all cells and faces, calls the respective workers
+    // on them, and then the copier function that puts things into the
+    // global matrix and right hand side. As an additional benefit,
+    // MeshWorker::mesh_loop() does all of this in parallel, using
+    // as many processor cores as your machine happens to have.
```

Comment:
If the data structures (vector and matrix) are set the right way...

Reply:
Sure, but that's not necessary to say, is it?